libmscore/tuplet.cpp

```diff
@@ -694,7 +694,8 @@ void Tuplet::read(XmlReader& e)
             else if (!DurationElement::readProperties(e))
                   e.unknown();
             }
-      Fraction f(_ratio.reduced().denominator(), _baseLen.fraction().denominator());
+      Fraction r = (_ratio == 1) ? _ratio : _ratio.reduced();
+      Fraction f(r.denominator(), _baseLen.fraction().denominator());
```

Comment:
I think this will still result in something strange.
Example:
1st measure 4/4, following measures 2/4. Type a dotted 1/4 rest in the first beat of the 1st measure, as in the example file (rename to mscz):

[test_tuplet.zip](https://github.com/musescore/MuseScore/files/577760/test_tuplet.zip)

Select this dotted quarter rest; Notes->Tuplets->Other... and set ratio to 8/6, i.e. 8 16th notes instead of 6 16th notes in the 3/8 duration of the tuplet.
Select the first measure and copy. Try to paste it into the second (2/4 time signature) measure.

I strongly think that the baselen should be correctly set for the tuplet instead of changing all the surrounding code to patch the behavior due to an incorrectly set baselen.
In principle a X:Y tuplet means X played notes in the space normally occupied by Y notes: it means that in the duration Z1/Z2 of the tuplet exactly Y notes should fit. The baselen of the tuplet is therefore (Z1/Z2) / Y ,i.e. Z1/Z2 x 1/Y. Then the duration is X x baselen = Y x (Z1/Z2) x 1/Y = Z1/Z2.

I think that tuplets imported from MusicXml would probably suffer from problems during importing if they sapn over the whole measure and their ratio is a not reduced fraction.